src/argilla/client/feedback/dataset/base.py

```diff
@@ -52,8 +58,8 @@ class FeedbackDatasetBase(ABC, HuggingFaceDatasetMixin):
     def __init__(
         self,
         *,
-        fields: List[AllowedFieldTypes],
-        questions: List[AllowedQuestionTypes],
+        fields: Union[List["AllowedFieldTypes"], List["AllowedRemoteFieldTypes"]],
+        questions: Union[List["AllowedQuestionTypes"], List["AllowedRemoteQuestionTypes"]],
```

Comment:
Here we still need to add the `Dict[str, Any]`

Reply:
maybe we can tackle this in another PR, I though it was just adding the type hints, but I've seen that we would also need to add the logic to transforms the `dict`s to the `pydantic.BaseModel`s